### ISSUE TITLE
Devise default commands registration mechanism

### ISSUE THREAD

**Issue by matiduck** [inausoft/netCLI#9]
This issue covers decision how we wanted to handle default commands (like help or version command). Crucial decisions are: 
1. Should we allow user to write own implementations of those commands
2. Registration of the commands - via `Map<T,Z>` method, via custom method (ex. `MapHelpCommand()` or other option? 
3. Should we make some bundle method (ex. `MapDefaultCommands`) which register all the default commands with different parameters?

**Comment by MarcinIN** [comment]
I would left a choice to the user. So registering both `help` and `version` or better yet `info` commands should be optional.
I have deleted `MapHelpCommand` extension method as one can simply call Map<T, Z> method to achieve the same results - I wouldn't add any additional overlay there.

For me the question is how should default registration look like?

```cs

services.AddCLI(config => {
                config.Map<MoveCommand, MoveCommandHandler>(true);
            });
```
or
```cs
services.AddCLI(config => {
                config.MapDefault<MoveCommand, MoveCommandHandler>();
            });
```

**Comment by jmarlew** [comment]
We had an interesting discussion with @matiduck and here's what I think about applying default commands.

First let's understand what default commands do we expect to support as "default"? I can imagine `help` (explains all available commands with their parameters) and `version` (lists executing assembly version). Any other ideas? Even if so, probably there won't be many of them.

I would see enabling these default behaviours by writing:

```cs
var config = new CLIConfiguration()
    .MapDefaultHelpCommand()
    .MapDefaultVersionCommand()
    .Map<MyCommand, MyCommandHandler>()
    ...
```

If I want to use my custom help command (which I may not want to be called with `help` keyword) I just write:

```cs
var config = new CLIConfiguration()
    .Map<MyHelpCommand, MyHelpCommandHandler>()
    .MapDefaultVersionCommand()
    .Map<MyCommand, MyCommandHandler>()
    ...
```

If I don't want any help or version commands - I just don't add `MapDefault...` calls.

**Comment by jmarlew** [comment]
Also I don't understand the part quoted below.

* Why are we talking "default" commands and the example is based on custom user command - `MoveCommand`? 
* What is the purpose of boolean parameter in first call?
* How does `Map` differ from `MapDefault` in these examples?

> For me the question is how should default registration look like?
> 
> ```cs
> services.AddCLI(config => {
>                 config.Map<MoveCommand, MoveCommandHandler>(true);
>             });
> ```
> 
> or
> 
> ```cs
> services.AddCLI(config => {
>                 config.MapDefault<MoveCommand, MoveCommandHandler>();
>             });
> ```

**Comment by MarcinIN** [comment]
@jmarlew I think any command can be set as default - even when it has `required` options.
So you would be perfectly fine writing:
```
config.MapDefault<HelpCommand, HelpCommandHandler>();
```

What I don't understand in the above examples is how would I be able to register (map) my own `HelpHandler` as default? 

Assuming that `.MapDefaultHelpCommand()` will map existing HelpHandler.

**Comment by MarcinIN** [comment]
It might not been clear but I would assume that mapping `MoveCommand` as default would result in calling:
`> myapp.exe move --from "C:\SamplePath" --to "D:\OthePath"`
to works just as fine as calling
`myapp.exe --from "C:\SamplePath" --to "D:\OthePath"`

**Comment by jmarlew** [comment]
OK, I thought that we are talking about default help or version commands. But it seems we are talking about single default command only. So it's clear that I should be able to register only one default command. So subsequent calls to `MapDefault` should cause an exception?

**Comment by MarcinIN** [comment]
`InvalidOperationException` probably - yes.

**Comment by MarcinIN** [comment]
@matiduck since it's last change needed for v2.0 - would you implement it - or should I do it?

**Comment by MarcinIN** [comment]
I will continue it from now.

**Comment by MarcinIN** [comment]
Resolved with #30
